libpimeval/src/pimPerfEnergyModels.cpp

```diff
@@ -371,8 +368,7 @@ pimParamsPerf::getPerfEnergyForFunc2(PimCmdEnum cmdType, const pimObjInfo& obj)
       double totalGDLOverhead = m_tGDL * numGDLItr * 2; // one read can be pipelined
       msRuntime = 2 * m_tR + m_tW + totalGDLOverhead + maxElementsPerRegion * m_blimpCoreLatency * numberOfOperationPerElement;
       msRuntime *= numPass;
-      mjEnergy = m_eAP * 3 * m_numChipsPerRank * numRanks + (m_eGDL * numGDLItr * 3 + (maxElementsPerRegion * m_blimpArithmeticEnergy * numberOfOperationPerElement)) * obj.getNumCoresUsed();
-      mjEnergy *= numPass;
+      mjEnergy = ((m_eAP * 3) + (m_eGDL * numGDLItr * 3 + (maxElementsPerRegion * m_blimpArithmeticEnergy * numberOfOperationPerElement))) * numCoresUsed * numPass;
```

Comment:
If this is bank-level energy, I actually feel like nothing here should be multiplied with numCores, right? It should be multiplied by the #ALU/bank * #bank

Reply:
Shouldn't we multiply by num of banks?

Reply:
I thought the numCores is the the number of subarrays, right? If this is bank-level ALU, then I think it has nothing to do with the number of subarrays.

Reply:
I forgot it, but could it be that numCore represents #banks in the bank-level architecture? Is that the reason you wrote the code in that way? I forgot if you mentioned this, but I just want to make sure we are using numCores correctly here

Reply:
yes numcores = numbanks for bank-level